docs/decisions/0004-explicitly-declare-enterprise-for-role-assignments.rst

```diff
@@ -0,0 +1,59 @@
+Enterprise role assignments will record a customer UUID
+-------------------------------------------------------
+
+Status
+======
+
+Proposed - January 2021
+
+Context
+=======
+
+The ``SystemWideEnterpriseUserRoleAssignment`` model is used to grant role-based permissions
+to enterprise users.  These permissions generally control the users' access to data
+and ability to perform certain functions within an enterprise customer.  This assignment model does `not` currently
+refer to any specific customer - the "context" to which a role is applied is inferred from
+a user (an ``auth.User`` record) having been "linked" to a customer via the ``EnterpriseCustomerUser`` model.
+
+However, a single ``auth.User`` may be linked to multiple ``EnterpriseCustomers``, and if that user
+is assigned a role (such as ``enterprise_admin``), it is inferred that the role assignment applies
+to **every** customer to which the user is linked.  This is undesirable; we generally do not want an admin
+of `Customer A` to also have admin permissions for `Customer B` (to which the user may be linked as a mere ``learner``).
+Instead, the ``SystemWideEnterpriseUserRoleAssignment`` model should explicitly record *which* customer
+the assigned user has the given role in.
+
+Decisions
+=========
+
+Add an ``enterprise_customer`` field to the assignment model
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+* It will be an optional field (for now?)...will ``null`` be a special/wildcard value?.
+* Can we ever make this a required field?  Specifically, will edX/internal users be problematic?
+
+We'll backfill the values of this field
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+* It's fine to have multiple assignments for the ``enterprise_learner`` role on a single user -
```

Comment:
Yes, it is also fine to have multiple assignments for the `enterprise_admin` role on a single user.

Reply:
Yup, fine from a data perspective.  I really meant, how can we know the ground truth of whether an admin of Customer A should be an admin in Customer B, too, or just a learner?  And I think we can't without getting a list of every admin user who belongs to multiple enterprises and doing a hand-audit.